helpers.js

```diff
@@ -0,0 +1,124 @@
+'lang sweet.js';
+
+const TypeCodes = {
+  Identifier: 0,
+  Keyword: 1,
+  Punctuator: 2,
+  NumericLiteral: 3,
+  StringLiteral: 4,
+  TemplateElement: 5,
+  Template: 6,
+  RegExp: 7
+};
+
+function check(obj, type, val) {
+  let isType = obj &&
+    obj.type === 'RawSyntax' &&
+    obj.value.token.typeCode === type;
+
+  if (isType && typeof val !== 'undefined') {
+    if (obj.value.token.typeCode === TypeCodes.StringLiteral) {
+      return obj.value.token.str === val;
+    }
+    return obj.value.token.value === val;
+  }
+  return isType;
+}
+
+export function unwrap(obj) {
+  let hasTok = obj && obj.value && obj.value.token;
+  if (hasTok && obj.value.token.typeCode === TypeCodes.StringLiteral) {
+    return {
+      value: obj.value.token.str
+    };
+  } else if (hasTok && obj.value.token.typeCode !== TypeCodes.Template) {
+    return {
+      value: obj.value.token.value
+    };
+  } else if (hasTok && obj.value.token.typeCode === TypeCodes.Template) {
+    return {
+      value: obj.value.token.items
+    };
+  } else if (obj && obj.type === 'RawDelimiter') {
+    return {
+      value: obj.inner
+    };
+  }
+  return {};
+}
+
+export function isIdentifier(obj, val) {
+  return check(obj, TypeCodes.Identifier, val);
+}
+
+export function fromIdentifier(obj, x) {
+  return obj.value.fromIdentifier(x);
+}
+
```

Comment:
Yeah I don't want to increase the API surface.

But you're right the overloading of `is*` is problematic. How about doing what you suggested and restrict `is*` to a single argument and then let the equality stuff be done by the consumer:

```js
const isFor = x => isKeyword(x) && unwrap(x).value === 'for';
const equalsIdentifier = (x, obj) => isIdentifier(obj) && unwrap(obj).value === x
```